ca/certificates.go

```diff
@@ -342,8 +345,17 @@ func NewRootCA(certBytes, keyBytes []byte, certExpiry time.Duration) (RootCA, er
 	// Create a Pool with all of the certificates found
 	pool := x509.NewCertPool()
 	for _, cert := range parsedCerts {
+		switch cert.SignatureAlgorithm {
+		case x509.SHA256WithRSA, x509.SHA384WithRSA, x509.SHA512WithRSA, x509.ECDSAWithSHA256, x509.ECDSAWithSHA384, x509.ECDSAWithSHA512:
+			break
+		default:
+			return RootCA{}, fmt.Errorf("unsupported signature algorithm: %s", cert.SignatureAlgorithm.String())
```

Comment:
You mean returning the error?

Mostly it was just easier because I wanted to use This function to help validate root certs submitted by users.

You are right in that we could just ignore the certs.  Behavior then is slightly more confusing in that if the first cert has to match the key, is it he first valid cert or the first provided cert?

I was also wondering whether it'd make sense to take the key and matching very separately rather than rely on the order.